src/error_hints.jl

```diff
@@ -1,7 +1,7 @@
 # provided by https://github.com/JuliaLang/julia/pull/35094
 function register_hints()
-    if VERSION >= v"1.5.0-DEV.491"
-        register_error_hint(MethodError) do io, exc, argtypes, kwargs
+    if isdefined(Base.Experimental, :register_error_hint)
```

Comment:
`Base.Experimental` module doesn't exist until julia-1.3, so we might still want to keep the version check here.

Reply:
Thanks @johnnychen94. I decided to check for the existence of `Experimental` instead. The thing I like about this approach is it's essentially a direct test of "will the following code work?" Version checks are a bit more indirect, and if you want to understand the meaning of the particular version you might have to do some git archeology (though a comment with hyperlink can fix that). Of course, they're extremely useful, especially in cases where there isn't a good alternative way to check.